### PR TITLE
optimize minicpm3 again

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

padding minicpm3's v cache to use sdp optimization

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/models/minicpm3.py [hunk 1]
```diff
@@ -20,6 +23,72 @@ def pre_compute_inv_freq(module: torch.nn.Module):
...
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        use_cache: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ):
+        # IPEX-LLM OPT: kv cache and quantize kv cache and sdp
+        inputs = input_ids if input_ids is not None else inputs_embeds
+        use_cache = use_cache if use_cache is not None else self.config.use_cache
+        use_cache = True if inputs.device.type == "xpu" else use_cache
```
Comment (rnwang04) [inline python/llm/src/ipex_llm/transformers/models/minicpm3.py:68]: For me, it's strange that force `use_cache` to True for xpu, it broke the original usage of this parameter.

**Review by rnwang04 — APPROVED** [review]
others LGTM.

File: python/llm/src/ipex_llm/transformers/models/minicpm3.py (hunk 1 above)
Comment (MeouSker77) [reply python/llm/src/ipex_llm/transformers/models/minicpm3.py:68]: if `use_cache=False`, `sdp_causal` will get wrong result, so set it to `True` automatically if using `sdp_causal`
